sympy/solvers/solvers.py

```diff
@@ -2653,22 +2655,58 @@ def _tsolve(eq, sym, **flags):
                     return _solve(lhs - rhs, sym, **flags)
             elif sym not in lhs.exp.free_symbols:
                 return _solve(lhs.base - rhs**(1/lhs.exp), sym, **flags)
-            elif not rhs and sym in lhs.exp.free_symbols:
-                # f(x)**g(x) only has solutions where f(x) == 0 and g(x) != 0 at
-                # the same place
-                sol_base = _solve(lhs.base, sym, **flags)
-                if not sol_base:
-                    return sol_base  # no solutions to remove so return now
-                return list(ordered(set(sol_base) - set(
-                    _solve(lhs.exp, sym, **flags))))
-            elif (rhs is not S.Zero and
-                        lhs.base.is_positive and
-                        lhs.exp.is_real):
-                return _solve(lhs.exp*log(lhs.base) - log(rhs), sym, **flags)
-            elif lhs.base == 0 and rhs == 1:
-                return _solve(lhs.exp, sym, **flags)
             else:
-                raise NotImplementedError
+                # sym in lhs.exp.free_symbols
+                if not rhs:
+                    # f(x)**g(x) only has solutions where f(x) == 0 and g(x) != 0 at
+                    # the same place
+                    sol_base = _solve(lhs.base, sym, **flags)
+                    if not sol_base:
+                        return sol_base  # no solutions to remove so return now
+                    return list(ordered(set(sol_base) - set(
+                    _solve(lhs.exp, sym, **flags))))
+                elif rhs.is_Rational or lhs.exp.is_real:
+                    sol=[]
+                    if lhs.base.is_real and lhs.base<0:
+                        sol.extend(_solve(-abs(lhs.base)**lhs.exp - rhs, sym, **flags) + _solve(abs(lhs.base)**lhs.exp - rhs, sym, **flags))
+                    if lhs.base.is_positive:
+                        if lhs.exp.is_real:
+                            sol.extend(_solve(lhs.exp*log(lhs.base) - log(rhs), sym, **flags))
+                        else:
+                            rewrite = lhs.rewrite(exp)
+                            if rewrite != lhs:
+                                sol.extend(_solve(rewrite - rhs, sym, **flags))
+                    if rhs.is_Rational:
+                        b_pos=[]
+                        b_negi=[]
+                        e0=[]
```

Comment:
when editing, please tidy up by putting space around `=`, `!=`, `==`, `+`, `-` and after commas.

Reply:
I did a bit of this while reviewing. Please check and also see if the XXX marked code is needed.

Reply:
> regarding `powdenest` and `sqrtdenest`...if you can think of a case where the one works when the other doesn't, then use the one that works.

I think there can be a possibility , okk i will search for that case.